### PR TITLE
feat: adding support to error handlers 

### PR DESCRIPTION
**Pull request by LeandroFranciscato**
## Related Issue or Design Document

Design Document issue:
https://github.com/ory/oathkeeper-maester/issues/69

stale old issue:
https://github.com/ory/oathkeeper-maester/issues/41

user slack question with no answer:
https://ory-community.slack.com/archives/C01340V8KSM/p1681971583368289

Oathkeeper errror handlers doc:
https://www.ory.sh/docs/oathkeeper/pipeline/error

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: api/v1alpha1/rule_types_test.go [hunk 1]
```diff
@@ -58,6 +58,14 @@ var (
           ]
         }
       }
+    ],
```
Comment (Demonsthere) [inline api/v1alpha1/rule_types_test.go:61]: Again this part would benefit greatly from using https://pkg.go.dev/embed

**Review by Demonsthere — APPROVED** [review]
Thank you very much for your contribution! We know how frustrating it can be to have your issues/requests hang in a limbo but we are currently too understaffed to be able to actively update the project 😞. 
I have no issues with the PR and we could merge/release it as is, but i will leave a few possible improvements that could make this more future proof.

File: tests/integration/rules_test.go [hunk 2]
```diff
@@ -418,7 +418,12 @@ func getRule1Json() string {
 				        }
 				    }
 			    }
-            ]
+            ],
```
[the remark this answers is not in the document]
Comment (LeandroFranciscato) [reply tests/integration/rules_test.go]: Done! good idea btw.

File: api/v1alpha1/rule_types_test.go [hunk 3]
```diff
@@ -424,11 +469,11 @@ func TestFilterNotValid(t *testing.T) {
...
 }
 
-func newStaticRule(authenticators []*Authenticator, authorizer *Authorizer, mutators []*Mutator) *Rule {
-	return newRule("r1", "test", "", "", newStringPtr(""), nil, newBoolPtr(false), authenticators, authorizer, mutators)
+func newStaticRule(authenticators []*Authenticator, authorizer *Authorizer, mutators []*Mutator, errors []*Error) *Rule {
+	return newRule("r1", "test", "", "", newStringPtr(""), nil, newBoolPtr(false), authenticators, authorizer, mutators, errors)
 }
 
-func newRule(name, namespace, upstreamURL, matchURL string, stripURLPath, configMapName *string, preserveURLHost *bool, authenticators []*Authenticator, authorizer *Authorizer, mutators []*Mutator) *Rule {
+func newRule(name, namespace, upstreamURL, matchURL string, stripURLPath, configMapName *string, preserveURLHost *bool, authenticators []*Authenticator, authorizer *Authorizer, mutators []*Mutator, errorrs []*Error) *Rule {
```
[the remark this answers is not in the document]
Comment (LeandroFranciscato) [reply api/v1alpha1/rule_types_test.go]: done

File: api/v1alpha1/rule_types_test.go [hunk 4]
```diff
@@ -443,6 +488,7 @@ func newRule(name, namespace, upstreamURL, matchURL string, stripURLPath, config
 		Authenticators: authenticators,
 		Authorizer:     authorizer,
 		Mutators:       mutators,
+		Errors:         errorrs,
```
[the remark this answers is not in the document]
Comment (LeandroFranciscato) [reply api/v1alpha1/rule_types_test.go]: done

File: api/v1alpha1/rule_types_test.go (hunk 1 above)
Comment (LeandroFranciscato) [reply api/v1alpha1/rule_types_test.go]: Also changed the integration tests to use embed.

**Review by Demonsthere — APPROVED** [review]
🙏

**Comment by Demonsthere** [comment]
The release CI failed due to outdated github credentials, but the image was pushed to dockerhub :) https://hub.docker.com/layers/oryd/oathkeeper-maester/v0.1.8/images/sha256-4ad8cacfff71d9cd54a4247e91e2bbc8be7ccc953b2569fa2d4cd0dc68ea152e?context=explore
We can now bump the image and CRD in the k8s repo

**Comment by LeandroFranciscato** [comment]
Cool.
I'm going to open the PR there.
Thank you.

**Pull request merged**